test/e2e/lua/dynamic_configuration.go

```diff
@@ -153,7 +159,14 @@ var _ = framework.IngressNginxDescribe("Dynamic Configuration", func() {
 		_, err = f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.IngressController.Namespace).Update(ingress)
 		Expect(err).ToNot(HaveOccurred())
 
-		time.Sleep(5 * time.Second)
+		By("generating the respective ssl listen directive")
```

Comment:
by moving `WaitForNginxServer` we can get rid of sleep and let `WaitForNginxServer` to take care of it